### PR TITLE
Rename duplicate test function names

### PR DESCRIPTION
**Pull request by akayjoshi**
## Description
Renamed duplicate test functions with different code
and Removed duplicate test with same code

## Phabricator Ticket
https://phabricator.wikimedia.org/T282460

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/applications/tests.py [hunk 1]
```diff
@@ -1840,23 +1840,14 @@ def _base_test_deleted_object_visibility(self, url, view, queryset):
...
             )
 
-    def test_list_object_visibility(self):
-        url = reverse("applications:list")
-        queryset = Application.objects.filter(
-            status__in=[Application.PENDING, Application.QUESTION]
-        )
-        self._base_test_deleted_object_visibility(
-            url, views.ListApplicationsView, queryset
-        )
```
Comment (jsnshrmn) [inline TWLight/applications/tests.py:1850]: I'm not actually seeing where else this queryset is being run against `_base_test_deleted_object_visibility`; this looks to me like we'd be losing test coverage.

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
Thanks for your work on this! The method renames look good, but I do have an inline comment on the method you deleted.

File: TWLight/applications/tests.py (hunk 1 above)
Comment (akayjoshi) [reply TWLight/applications/tests.py]: yes you were right I mistakenly deleted that thinking its duplicate, I have rewrote it now

Comment (jsnshrmn) [reply TWLight/applications/tests.py]: Thanks! These method names are a big improvement in descriptiveness in addition to removing the naming collisions.

**Review by jsnshrmn — APPROVED** [review]

**Pull request merged**